src/devices/GrovePi/GrovePi.cs

```diff
@@ -165,27 +175,20 @@ public void WriteCommand(GrovePiCommand command, GrovePort pin, byte param1, byt
         public PinValue DigitalRead(GrovePort pin)
         {
             WriteCommand(GrovePiCommand.DigitalRead, pin, 0, 0);
-            byte tries = 0;
-            IOException? innerEx = null;
-            // When writing/reading to the I2C port, GrovePi doesn't respond on time in some cases
-            // So we wait a little bit before retrying
-            // In most cases, the I2C read/write can go thru without waiting
-            while (tries < MaxRetries)
+            try
             {
-                try
+                var data = ReadCommand(GrovePiCommand.DigitalRead, pin);
+                if (data is null || data.Length < 2)
                 {
-                    return (PinValue)_i2cDevice.ReadByte();
-                }
-                catch (IOException ex)
-                {
-                    // Give it another try
-                    innerEx = ex;
-                    tries++;
-                    Thread.Sleep(10);
+                    return (PinValue)(-1);
                 }
-            }
 
-            throw new IOException($"{nameof(DigitalRead)}: Failed to read byte with command {GrovePiCommand.DigitalRead}", innerEx);
```

Comment:
Why are you removing the throw here and replace with the -1 logic? Throwing is a good way to say that things did not work properly.

Reply:
@Ellerbach 
I initially considered making this change because the existing [`AnalogRead` method](https://github.com/dotnet/iot/pull/2359/files#diff-499d0f2c3b3d578696ec4fefd1a30c638568203e461f54021ef624576baf384eL210) was designed to return -1 instead of throwing an exception. However, I have adjusted the behavior at commit 93c630371654591b84a5d69241ad368607f2acaa accordingly.